### DISCUSSION TITLE
ftplugin in fennel?

### DISCUSSION THREAD

**Discussion by ajayvigneshk** [rktjmp/hotpot.nvim#41] [Q&A] [answered]
Hey! Just curious, is it possible to write ftplugin configs in fennel and have hotpot compile it into lua?

**Comment by bangedorrunt** [comment]
I don't think it's feasible with Hotpot at the moment. You can use Fennel binary file to compile it manually though.

**Reply by rktjmp** [reply]
This is correct, because hotpot acts as a lua loader, it doesn't really interact with the ftplugins dir, as neovim doesn't consult the regular lua loader path for those. What you could do is something like

```
fnl/
  /my-config
    /ftplugins
      /elixir.fnl
/ftplugin
  /elixir.lua <- "require("my-config.ftplugins.elixir")
```

Not great but it should work and once the lua bytecode loader stuff is worked out it's probably no slower really.

**Reply by ajayvigneshk** [reply]
Thanks @rktjmp . Makes sense

**Reply by kejadlen** [reply]
Is there a way to do something like this manually to load the Fennel code?

https://github.com/neovim/neovim/blob/master/runtime/ftplugin.vim

**Reply by rktjmp** [reply]
That's a mighty big brain you've got there, not sure why I never thought to add a `FileType` autocommand.

```fennel
;; in your config
(let [{: nvim_create_autocmd : nvim_create_augroup} vim.api
      au-group (nvim_create_augroup :hotpot-ft {})
      cb #(pcall require (.. :ftplugin. (vim.fn.expand "<amatch>")))]
  (nvim_create_autocmd :FileType {:callback cb :group au-group}))
```

```fennel
;; ~/.config/nvim/ftplugin/wow.fnl
(print :wow!)
```

```viml
"in vim
:set ft=wow
" wow!
```

For now you can drop that or similar in your config and I will mainline it soon. It may end up behind a flag as there is perhaps some performance implications there when we search the RTP twice, once for us, then again for the normal stuff. Searching the RTP _can_ be slow.

**Reply by kejadlen** [reply]
Excellent, thank you!

**Comment by rktjmp** [comment, marked as answer]
```fennel
;; in your config
(let [{: nvim_create_autocmd : nvim_create_augroup} vim.api
      au-group (nvim_create_augroup :hotpot-ft {})
      cb #(pcall require (.. :ftplugin. (vim.fn.expand "<amatch>")))]
  (nvim_create_autocmd :FileType {:callback cb :group au-group}))
```

```fennel
;; ~/.config/nvim/ftplugin/wow.fnl
(print :wow!)
```

```viml
"in vim
:set ft=wow
" wow!
```

Repost for selectable answer as the comment ends up in a loading break.

https://github.com/rktjmp/hotpot.nvim/discussions/41#discussioncomment-3050564

**Comment by rktjmp** [comment]
Discussion not relevant to v2. Support is built in.
